code/game/machinery/cryo.dm

```diff
@@ -202,22 +233,39 @@
 				var/heal_brute = occupant.getBruteLoss() ? min(1, 20/occupant.getBruteLoss()) : 0
 				var/heal_fire = occupant.getFireLoss() ? min(1, 20/occupant.getFireLoss()) : 0
 				occupant.heal_limb_damage(heal_brute,heal_fire)
-		var/has_cryo = occupant.reagents.get_reagent_amount("cryoxadone") >= 1
-		var/has_clonexa = occupant.reagents.get_reagent_amount("clonexadone") >= 1
-		var/has_cryo_medicine = has_cryo || has_clonexa
-		if(beaker && !has_cryo_medicine)
-			beaker.reagents.trans_to(occupant, 1, 10)
+
+	// Chemical healing if cryo meds are involved
+	if(beaker && occupant.reagents && beaker.reagents)
+		var/occupant_has_cryo_meds = occupant.reagents.get_reagent_amount("cryoxadone") >= 1 || occupant.reagents.get_reagent_amount("clonexadone") >= 1
+		var/beaker_has_cryo_meds = beaker.reagents.get_reagent_amount("cryoxadone") >= 1 || beaker.reagents.get_reagent_amount("clonexadone") >= 1
+
+		// To administer, either the occupant has cryo meds and the beaker doesn't or vice versa (not both)
+		var/can_administer = (occupant_has_cryo_meds ^ beaker_has_cryo_meds) && length(beaker.reagents.reagent_list)
+		if(can_administer && occupant_has_cryo_meds)
+			// If its the case of the occupant has cryo meds and not the beaker, we need to pace out the dosage
+			// So lets make sure they don't already have some of the beaker drugs
+			for(var/datum/reagent/cur_beaker_reagent in beaker.reagents.reagent_list)
+				for(var/datum/reagent/cur_occupant_reagent in occupant.reagents.reagent_list)
+					if(cur_beaker_reagent.id == cur_occupant_reagent.id)
+						can_administer = FALSE
+						break
+
+		if(can_administer)
+			beaker.reagents.trans_to(occupant, 5)
 			beaker.reagents.reaction(occupant)
-		if(!occupant.getBruteLoss(TRUE) && !occupant.getFireLoss(TRUE) && !occupant.getCloneLoss() && autoeject) //release the patient automatically when brute and burn are handled on non-robotic limbs
-			display_message("external wounds are")
+
+	if(autoeject)
+		//release the patient automatically when brute and burn are handled on non-robotic limbs
+		if(!occupant.getBruteLoss(TRUE) && !occupant.getFireLoss(TRUE) && !occupant.getCloneLoss())
+			display_message("Patient's external wounds are healed.")
 			go_out(TRUE)
 			return
-		if(occupant.health >= 100 && autoeject)
-			display_message("external wounds are")
+		if(occupant.health >= 100)
```

Comment:
i think it would be wiser to use occupant.health >= occupant.maxhealth JUST in case.